node/lib/util/hook.js

```diff
@@ -43,17 +44,26 @@ const process   = require("process");
  * @param {String[]} args
  * @return {String}
  */
-exports.execHook = co.wrap(function *(repo, name, args=[]) {
+exports.execHook = co.wrap(function(repo, name, args=[]) {
     assert.isString(name);
 
     const rootDirectory = repo.path();
     const hookPath = path.join(rootDirectory, "hooks");
     const absPath = path.resolve(hookPath, name);
 
+    if (!fs.existsSync(absPath)) {
+        return absPath + " hook does not exist.";
+    }
+
     try {
         process.chdir(repo.workdir());
-        const result = yield ChildProcess.execFile(absPath, args);
-        console.log(result.stdout);
+        const result = spawn(absPath, args);
+        result.stdout.on("data", function(data) {
+            console.log(data.toString());
```

Comment:
Doesn't this end up inserting additional newlines?

Reply:
And, actually, is there a way to just hook our stdout/stderr up to the process so that we don't have to read/write?